File: generated-code/swift/codecraft/TcpReadWrite/Sources/TransGenTest/Color.swift
/// RGBA Color
public struct Color {
    /// Red component
    let r: Float

    /// Green component
    let g: Float

    /// Blue component
    let b: Float

    /// Alpha (opacity) component
    let a: Float

    /// Read Color from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Color {
        var r: Float
        r = stream.readFloat()
        var g: Float
        g = stream.readFloat()
        var b: Float
        b = stream.readFloat()
        var a: Float
        a = stream.readFloat()
        return Color(r: r, g: g, b: b, a: a)
    }

    /// Write Color to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeFloat(r)
        stream.writeFloat(g)
        stream.writeFloat(b)
        stream.writeFloat(a)
    }
}

File: generated-code/kotlin/codecraft/FileReadWrite/src/main/kotlin/trans_gen_test/Color.kt
package trans_gen_test

import trans_gen_test.util.StreamUtil

/**
 * RGBA Color
 */
class Color {
    /**
     * Red component
     */
    var r: Float
    /**
     * Green component
     */
    var g: Float
    /**
     * Blue component
     */
    var b: Float
    /**
     * Alpha (opacity) component
     */
    var a: Float

    constructor(r: Float, g: Float, b: Float, a: Float) {
        this.r = r
        this.g = g
        this.b = b
        this.a = a
    }

    /**
     * Write Color to output stream
     */
    @Throws(java.io.IOException::class)
    fun writeTo(stream: java.io.OutputStream) {
        StreamUtil.writeFloat(stream, r)
        StreamUtil.writeFloat(stream, g)
        StreamUtil.writeFloat(stream, b)
        StreamUtil.writeFloat(stream, a)
    }

    /**
     * Get string representation of Color
     */
    override fun toString(): String {
        var stringBuilder = StringBuilder("Color { ")
        stringBuilder.append("r: ")
        stringBuilder.append(r)
        stringBuilder.append(", ")
        stringBuilder.append("g: ")
        stringBuilder.append(g)
        stringBuilder.append(", ")
        stringBuilder.append("b: ")
        stringBuilder.append(b)
        stringBuilder.append(", ")
        stringBuilder.append("a: ")
        stringBuilder.append(a)
        stringBuilder.append(" }")
        return stringBuilder.toString()
    }

    companion object {
        /**
         * Read Color from input stream
         */
        @Throws(java.io.IOException::class)
        fun readFrom(stream: java.io.InputStream): Color {
            var r: Float
            r = StreamUtil.readFloat(stream)
            var g: Float
            g = StreamUtil.readFloat(stream)
            var b: Float
            b = StreamUtil.readFloat(stream)
            var a: Float
            a = StreamUtil.readFloat(stream)
            return Color(r, g, b, a)
        }
    }
}